### PR TITLE
codeintel: Fix line translation in git diff edge cases

### PR DESCRIPTION
**Pull request by efritz**
**TODO**

## Test plan

**TODO**

### CODE REVIEW

File: internal/codeintel/codenav/gittree_translator_test.go [hunk 1]
```diff
@@ -388,8 +388,87 @@ var prometheusTestCases = []gitTreeTranslatorTestCase{
...
++func (h *Handle) routeHandle(route *Route, req *nl.NetlinkRequest, msg *nl.RtMsg) ([][]byte, error) {
++	if req.NlMsghdr.Type != unix.RTM_GETROUTE && (route.Dst == nil || route.Dst.IP == nil) && route.Src == nil && route.Gw == nil && route.MPLSDst == nil {
++		return nil, fmt.Errorf("Either Dst.IP, Src.IP or Gw must be set")
+ 	}
+
+ 	family := -1
+`
+
+var netlinkTestCases = []gitTreeTranslatorTestCase{
+	{netlinkDiff, "netlink", "", 656, false, 0},
```
Comment (efritz) [inline internal/codeintel/codenav/gittree_translator_test.go:467]: @Strum355 Not sure if this is the correct line number, should be the first diff hunk:

```
-	return h.routeHandle(route, req, nl.NewRtMsg())
+	_, err := h.routeHandle(route, req, nl.NewRtMsg())
```

This test fails and if that line is correct let's just fix the code!

Comment (Strum355) [reply internal/codeintel/codenav/gittree_translator_test.go:467]: This _particular_ one seems to work fine on dotcom https://sourcegraph.com/github.com/vishvananda/netlink/-/blob/route_linux.go?L655:27&subtree=true&popover=pinned#tab=references

Comment (efritz) [reply internal/codeintel/codenav/gittree_translator_test.go:467]: That def is search based 🤔

Comment (Strum355) [reply internal/codeintel/codenav/gittree_translator_test.go:467]: just noticed...thats extra weird

Comment (Strum355) [reply internal/codeintel/codenav/gittree_translator_test.go:467]: We wanna test this one btw https://sourcegraph.com/github.com/vishvananda/netlink/-/blob/route_linux.go?L670:30&popover=pinned

Comment (Strum355) [reply internal/codeintel/codenav/gittree_translator_test.go:467]: or this https://sourcegraph.com/github.com/vishvananda/netlink/-/blob/route_linux.go?L683:31&popover=pinned

Comment (efritz) [reply internal/codeintel/codenav/gittree_translator_test.go]: Updated test number (still failing, as expected)

Comment (efritz) [reply internal/codeintel/codenav/gittree_translator_test.go]: ...or maybe I'm off by one with my inputs...

**Pull request closed without merging**